django/db/models/fields/tuple_lookups.py

```diff
@@ -129,6 +130,20 @@ def as_sql(self, compiler, connection):
         return super().as_sql(compiler, connection)
 
 
+class TupleComparisonMixin:
+    def as_oracle(self, compiler, connection):
+        """
+        Regardless of whether a subquery has been limited to return
+        a single row, this error is raised for <, <=, >, and >=:
+        ORA-01796: this operator cannot be used with lists
+        """
```

Comment:
The version before this tried to guard for subqueries inside `get_fallback_sql()`, since I was seeing `Query object is not iterable` on `for val in self.rhs`, but even the versions of Oracle that support tuple lookups throw this ORA-01796 error, so I just decided to guard oracle. If there are other vendors with `get_fallback_sql()` methods, they may need adjusting to handle subqueries. Open to ideas here.

Reply:
I spent a bit of time with the latest versions of Oracle so I have no ideas to offer.